src/openforms/registrations/contrib/objects_api/migrations/0017_move_singleton_data.py

```diff
@@ -0,0 +1,89 @@
+# Generated by Django 4.2.11 on 2024-05-27 13:06
+
+from django.db import migrations
+
+from django.db.migrations.state import StateApps
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+
+def move_singleton_data_to_objects_api_set_model(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    ObjectsAPIConfig = apps.get_model("registrations_objects_api", "ObjectsAPIConfig")
+    ObjectsAPIGroupConfig = apps.get_model(
+        "registrations_objects_api", "ObjectsAPIGroupConfig"
+    )
+
+    solo_model = ObjectsAPIConfig.objects.first()
+    if not solo_model:
+        return
+
+    new_config_set = ObjectsAPIGroupConfig(
+        name="Objects APIs",
+        objects_service=solo_model.objects_service,
+        objecttypes_service=solo_model.objecttypes_service,
+        drc_service=solo_model.drc_service,
+        catalogi_service=solo_model.catalogi_service,
+        objecttype=solo_model.objecttype,
+        objecttype_version=solo_model.objecttype_version,
+        productaanvraag_type=solo_model.productaanvraag_type,
+        informatieobjecttype_submission_report=solo_model.informatieobjecttype_submission_report,
+        informatieobjecttype_submission_csv=solo_model.informatieobjecttype_submission_csv,
+        informatieobjecttype_attachment=solo_model.informatieobjecttype_attachment,
+        organisatie_rsin=solo_model.organisatie_rsin,
+        content_json=solo_model.content_json,
+        payment_status_update_json=solo_model.payment_status_update_json,
```

Comment:
so looking at the zgw api group, I notice here the models specify a default `objecttype` and `objecttype_version`, plus the `document_type` for the attachments, submission report and CSV. The equivalents for that don't exist on the ZGW API group - you must specify zaaktype/documenttype at the individual form level.

I think this applies for objecttype/objecttype_version for sure - there's no point IMO to provide those at the API group level. They are also easily migrated in this data migration from the solo-config to all the registration backend instances that don't have them specified explicitly.

For some of the other fields I think the answer is a bit harder:

* `productaanvraag_type` -> this is legacy objects API only, so I would tackle this with a data migration on the relevant registration backend instances. It definitely does not belong in the API group model
* `informatieobjecttype_submission_report`, `informatieobjecttype_submission_csv` and `informatieobjecttype_attachment` - these are tied to the catalogi API of the group, so I think it makes sense to configure them here, because the UX impact of every new registration backend with the same API group would become too bad, having to select the same document type over and over again
* `organisatie_rsin` also exists on zgw API group, can stay
* `content_json` and `payment_status_update_json` are only relevant for legacy objects API - I'd keep those on the solo model and remove them entirely in Open Forms 3.0 when we drop support for the legacy flavour. Then we can likely drop the solo-model entirely.

Reply:
The goal of such an API group should really be treat a set of connection details as one unit, and I suppose I can live with some configuration defaults that don't get overridden frequently. But the objecttype should really be a different object type for every form, so there's no point in providing a default for that and all the related implementation complexities.